core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -90,37 +90,46 @@ object ConsumerGroupCommand {
-      topicPartitions
-        .sortBy { case topicPartition => topicPartition.partition }
-        .foreach { topicPartition =>
-          describePartition(group, topicPartition.topic, topicPartition.partition, getPartitionOffset(topicPartition),
-            getOwner(topicPartition))
-        }
+                                         getMemberId: TopicAndPartition => Option[String],
+                                         getOwner: TopicAndPartition => Option[String],
+                                         getOwnerHost: TopicAndPartition => Option[String]): Unit = {
+      if (topicPartitions.isEmpty)
+        describePartition(group, None, None, None, getMemberId(null), getOwner(null), getOwnerHost(null))
+      else
+        topicPartitions
+          .sortBy { case topicPartition => topicPartition.partition }
+          .foreach { topicPartition =>
+            describePartition(group, Some(topicPartition.topic), Some(topicPartition.partition), getPartitionOffset(topicPartition),
+              getMemberId(topicPartition), getOwner(topicPartition), getOwnerHost(topicPartition))
+          }
     }
 
     protected def printDescribeHeader() {
-      println("%-30s %-30s %-10s %-15s %-15s %-15s %s".format("GROUP", "TOPIC", "PARTITION", "CURRENT-OFFSET", "LOG-END-OFFSET", "LAG", "OWNER"))
+      println("%-30s %-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format("GROUP", "TOPIC", "PARTITION", "CURRENT-OFFSET", "LOG-END-OFFSET", "LAG", "MEMBER-ID", "OWNER", "OWNER-HOST"))
     }
 
     private def describePartition(group: String,
-                                  topic: String,
-                                  partition: Int,
+                                  topic: Option[String],
+                                  partition: Option[Int],
                                   offsetOpt: Option[Long],
-                                  ownerOpt: Option[String]) {
+                                  memberOpt: Option[String],
+                                  ownerOpt: Option[String],
+                                  ownerHostOpt: Option[String]) {
       def print(logEndOffset: Option[Long]): Unit = {
         val lag = offsetOpt.filter(_ != -1).flatMap(offset => logEndOffset.map(_ - offset))
-        println("%-30s %-30s %-10s %-15s %-15s %-15s %s".format(group, topic, partition, offsetOpt.getOrElse("unknown"), logEndOffset.getOrElse("unknown"), lag.getOrElse("unknown"), ownerOpt.getOrElse("none")))
+        println("%-30s %-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format(group, topic.getOrElse(""), partition.getOrElse(""), offsetOpt.getOrElse(""), logEndOffset.getOrElse(""), lag.getOrElse(""), memberOpt.getOrElse("none"), ownerOpt.getOrElse(""), ownerHostOpt.getOrElse("")))
       }
       getLogEndOffset(topic, partition) match {
         case LogEndOffsetResult.LogEndOffset(logEndOffset) => print(Some(logEndOffset))
+        case LogEndOffsetResult.NoTopicPartition => print(offsetOpt)
```

Comment:
If topicPartition doesn't exist, should we really pass in offsetOpt? It seems it's more intuitive to pass in None.